lib/Makefile

```diff
@@ -35,28 +35,28 @@ CCVER := $(shell $(CC) --version)
 # space-optimized defaults.
 ZSTD_LIB_MINIFY ?= 0
 ifneq ($(ZSTD_LIB_MINIFY), 0)
-	HAVE_CC_OZ ?= $(shell echo "" | $(CC) -Oz -x c -c - -o /dev/null 2> /dev/null && echo 1 || echo 0)
-	ZSTD_LEGACY_SUPPORT ?= 0
-	ZSTD_LIB_DEPRECATED ?= 0
-	HUF_FORCE_DECOMPRESS_X1 ?= 1
-	ZSTD_FORCE_DECOMPRESS_SHORT ?= 1
-	ZSTD_NO_INLINE ?= 1
-	ZSTD_STRIP_ERROR_STRINGS ?= 1
-	ifneq ($(HAVE_CC_OZ), 0)
-		# Some compilers (clang) support an even more space-optimized setting.
-		CFLAGS += -Oz
-	else
-		CFLAGS += -Os
-	endif
-	CFLAGS += -fno-stack-protector -fomit-frame-pointer -fno-ident \
-	          -DDYNAMIC_BMI2=0 -DNDEBUG
+  HAVE_CC_OZ ?= $(shell echo "" | $(CC) -Oz -x c -c - -o /dev/null 2> /dev/null && echo 1 || echo 0)
+  ZSTD_LEGACY_SUPPORT ?= 0
+  ZSTD_LIB_DEPRECATED ?= 0
+  HUF_FORCE_DECOMPRESS_X1 ?= 1
+  ZSTD_FORCE_DECOMPRESS_SHORT ?= 1
+  ZSTD_NO_INLINE ?= 1
+  ZSTD_STRIP_ERROR_STRINGS ?= 1
+ifneq ($(HAVE_CC_OZ), 0)
+    # Some compilers (clang) support an even more space-optimized setting.
+    CFLAGS += -Oz
 else
-	CFLAGS += -O3
+    CFLAGS += -Os
+endif
+  CFLAGS += -fno-stack-protector -fomit-frame-pointer -fno-ident \
+            -DDYNAMIC_BMI2=0 -DNDEBUG
+else
+  CFLAGS += -O3
```

Comment:
nit: The indentation is strange.

Reply:
It's probably not visible on github,
but the most important part of the indentation change is that
I switched `tab` for 2 `space`.

Reason is, a command or assignment preceded by a `tab` is considered part of ongoing recipe.

This can lead to strange consequences when introducing some new recipe before an assignment.

For example, consider this segment, using a `tab` : 
```
ifeq ($(CONDITION), $(VALUE))
	CFLAGS += $(FLAG)
endif
```

Now, let's introduce a recipe just before, even if it doesn't look like so : 
```
recipe: prereq
	$(COMMAND)

# some comment
# clearly this is a different section 

ifeq ($(CONDITION), $(VALUE))
	CFLAGS += $(FLAG)
endif
```

now, `CFLAGS += $(FLAG)` is part of `recipe` !

This confusion no longer happens when using `space` instead.

Reply:
But it looks like there are some tabs leftover on lines 47 & 49. And can the nested if-statement be indented 4-spaces?

Reply:
Ah, GitHub doesn't do a great job indicating that I was referring to the whole block of lines 38-54, not this particular line.

Reply:
Oh, I see there are now 4 spaces, which just happen to be identical to a tab. Can the `ifneq` line not have leading spaces?

Reply:
> Can the `ifneq` line not have leading spaces?

Do you mean, can the inner `ifneq` be indented, like the rest of the code block inside the outer `ifneq`?

I think it can.
It's just that the automatic parser didn't like it, and would mess up syntax coloring if I tried to indent the `ifneq`.
So I didn't feel it worth fighting it.

Reply:
Sounds good